apps/voting/test/voting.js

```diff
@@ -495,6 +495,47 @@ contract('Voting App', accounts => {
         })
     })
 
+    context('changing token supply', () => {
+        const holder1 = accounts[1]
+        const holder2 = accounts[2]
+
+        const neededSupport = pct16(50)
+        const minimumAcceptanceQuorum = pct16(20)
+
+        beforeEach(async () => {
+            token = await MiniMeToken.new(NULL_ADDRESS, NULL_ADDRESS, 0, 'n', 0, 'n', true) // empty parameters minime
+
+            await token.generateTokens(holder1, 1)
+            await token.generateTokens(holder2, 1)
+
+            await voting.initialize(token.address, neededSupport, minimumAcceptanceQuorum, votingTime)
+        })
+
+        it('uses the correct snapshot value if tokens are minted afterwards', async () => {
+            // Create vote and afterwards generate some tokens
+            const voteId = createdVoteId(await voting.newVote(EMPTY_SCRIPT, 'metadata'))
+            await token.generateTokens(holder2, 1)
+
+            const [isOpen, isExecuted, startDate, snapshotBlock, supportRequired, minQuorum, y, n, votingPower, execScript] = await voting.getVote(voteId)
+
+            // Generating tokens advanced the block by one
+            assert.equal(snapshotBlock.toString(), await getBlockNumber() - 2, 'snapshot block should be correct')
+            assert.equal(votingPower.toString(), (await token.totalSupplyAt(snapshotBlock)).toString(), 'total voters should be correct')
+        })
+
+        it('uses the correct snapshot value if tokens are minted in the same block', async () => {
+            // Create vote and generate some tokens in the same transaction
+            // Requires the voting mock to be the token's owner
+            await token.changeController(voting.address)
+            const voteId = createdVoteId(await voting.newTokenAndVote(holder2, 1, 'metadata'))
+
+            const [isOpen, isExecuted, startDate, snapshotBlock, supportRequired, minQuorum, y, n, votingPower, execScript] = await voting.getVote(voteId)
+
+            assert.equal(snapshotBlock.toString(), await getBlockNumber() - 1, 'snapshot block should be correct')
+            assert.equal(votingPower.toString(), (await token.totalSupplyAt(snapshotBlock)).toString(), 'total voters should be correct')
```

Comment:
We could even check that those values match the expected total number of tokens (3, I guess)

Reply:
Actually 2, since it's the block before, but 👍.